src/docfx/build/toc/TocMap.cs

```diff
@@ -179,13 +182,33 @@ private static (int subDirectoryCount, int parentDirectoryCount) GetRelativeDire
         var allTocs = new List<(FilePath file, HashSet<FilePath> docs, HashSet<FilePath> tocs, bool shouldBuildFile)>();
         var includedTocs = new HashSet<FilePath>();
         var allServicePages = new List<FilePath>();
+        var tocFilesFromBuildScope = _buildScope.GetFiles(ContentType.Toc);
+        var (originalReferenceTOCs, targetReferenceTOCs) = GetOriginalReferenceTocWithTargetReferenceToc(tocFilesFromBuildScope);
 
         // Parse and split TOC
         ParallelUtility.ForEach(
```

Comment:
Is it possible to merge these two loops?

Reply:
Will try

Reply:
Done